tools/cli/pkg/command/completion.go

```diff
@@ -0,0 +1,118 @@
+package command
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// CompletionCommand represents an execution of the kcp completion command
+type CompletionCommand struct {
+	cobraCmd       *cobra.Command
+	completeTarget string
+}
+
+var defaultFile string
+var validArguements = []string{"bash", "zsh", "fish", "powershell"}
+
+const errorMsg = "Error: accepts 1 arg(s), received 0 \n"
+const suggestionMsg = "Please use `kcp completion [bash|zsh|fish|powershell]`"
+const savedInMsg = "Saved in"
+const savedFileName = "kcp_completion"
+
+// NewCompletionCmd constructs a new instance of CompletionCommand and configures it in terms of a cobra.Command
+func NewCompletionCommand() *cobra.Command {
+	cmd := CompletionCommand{}
+	cobraCmd := &cobra.Command{
+		Use:   "completion [bash|zsh|fish|powershell]",
+		Short: "Generate completion script",
+		Long: `To load completions:
+
+Bash:
+
+$ source <(kcp completion bash)
+
+# To load completions for each session, execute once:
+Linux:
+  $ kcp completion bash > /etc/bash_completion.d/kcp
+MacOS:
+  $ kcp completion bash > /usr/local/etc/bash_completion.d/kcp
+
+Zsh:
+
+# If shell completion is not already enabled in your environment you will need
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ kcp completion zsh > "${fpath[1]}/_kcp"
+
+# You will need to start a new shell for this setup to take effect.
+
+Fish:
+
+$ kcp completion fish | source
+
+# To load completions for each session, execute once:
+$ kcp completion fish > ~/.config/fish/completions/kcp.fish
+
+Powershell:
+
+PS> kcp completion powershell | Out-String | Invoke-Expression
+
+# To load completions for every new session, run:
+PS> kcp completion powershell > kcp.ps1
+# and source this file from your powershell profile.
+`,
+		Example:               `kcp completion bash                            Display completions in bash.`,
+		DisableFlagsInUseLine: true,
+		ValidArgs:             validArguements,
+		Args:                  cobra.MaximumNArgs(1),
```

Comment:
This can be omitted, and `Validate()` could be used to check for exactly on argument, and with the suggestion message.

Reply:
removed the line of `Args:                  cobra.MaximumNArgs(1),`